graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java

```diff
@@ -171,14 +175,16 @@ private Map<String, Object> runQueryWithLowLevelClient(String query) throws Atla
     }
 
     private DirectIndexQueryResult performAsyncDirectIndexQuery(SearchParams searchParams) throws AtlasBaseException, IOException {
+        AtlasPerfMetrics.MetricRecorder metric = RequestContext.get().startMetricRecord("performAsyncDirectIndexQuery");
         DirectIndexQueryResult result = null;
+        boolean contextIdExists = StringUtils.isNotEmpty(searchParams.getSearchContextId()) && searchParams.getSearchContextSequenceNo() != null;
         try {
-            if(StringUtils.isNotEmpty(searchParams.getSearchContextId()) && searchParams.getSearchContextSequenceNo() != null) {
+            if(contextIdExists) {
                 // If the search context id and greater sequence no is present, then we need to delete the previous search context async
-                processRequestWithSameSearchContextId(searchParams);
+                    processRequestWithSameSearchContextId(searchParams);
             }
             AsyncQueryResult response = submitAsyncSearch(searchParams, false).get();
-
+            //Sleep for 5 seconds to allow ES to process the request
```

Comment:
Where is the sleep for 5 seconds implemented? Is it absolutely necessary?

Reply:
Removed this test comment